submissions/algaroba.json

```diff
@@ -0,0 +1,19 @@
+{
+    "name": "Algaroba",
+    "contributors": [
+        { "name": "Amar Shah", "website": "https://amarshah1.github.io/" },
+        { "name": "Federico Mora", "website": "https://federico.morarocha.ca/" },
+        { "name": "Sanjit Seshia", "website": "https://people.eecs.berkeley.edu/~sseshia/" }
+    ],
+    "contacts": ["Amar Shah <amarshah@cmu.edu>"],
+    "archive": {
+        "url": "https://github.com/uclid-org/algaroba/archive/refs/tags/v1.0.0.tar.gz"
+    },
+    "website": "https://github.com/uclid-org/algaroba",
+    "system_description": "https://github.com/uclid-org/algaroba/files/15255360/Algaroba_at_the_2024_SMTCOMP.pdf",
+    "command": ["algaroba"],
```

Comment:
@amarshah1 Thanks for submitting to SMT-COMP! I have tested the submission on our infrastructure and it seems that the specified binary `algaroba` is not present in the archive. The archive contains folder `algaroba-1.0.0/bin/` with contents `algaroba.ml` and `dune`. I am doing anything wrong or is the binary indeed not in the archive?

If you will be fixing the command, please also use the relative path to the executable given that the archive is unpacked to the current working directory (in your case by calling `tar -xzf v1.0.0.tar.gz`). For the current archive, the command should probably be something like `algaroba-1.0.0/bin/algaroba`.

Reply:
Hi Martin,

Thanks for your reply! I am a little confused about how I should do this. What is the environment that SMTCOMP will be run on, so that I can create an executable that runs on that environment?

Reply:
You can find more details about the execution environment in the Final call for solvers and on the [SMT-Comp website](https://smt-comp.github.io/2024/specs/):

> It is also possible to locally emulate and test the computing environment on the competition machines using the following instructions:
> https://gitlab.com/sosy-lab/benchmarking/competition-scripts/#computing-environment-on-competition-machines

Basically, the machines are running Ubuntu 22.04 with a few extra packages (for details, see the link above). You can test your solver using the referenced Dockerfile, which should be almost identical to the machines used. If you rebase your pull request to the latest version of `master`, your solver will also be executed in this environment as a part of CI check.

Reply:
Thank you! I couldn't get the Docker working, but I compiled the executable on Ubuntu 22.04.  I have updated the link

Reply:
I just committed again with an executable compiled in that computing environment, but it still seems to fail the tests. Do you have any insight into what is happening? Do you know what the trivial_bench files look like? I can try running them locally

Reply:
Is there a way to run a script once at the beginning, rather than having it run a script every time the solver is used?

Reply:
Ok, I got it to pass all of the checks. Thank you for all your help!